From e2864c8156e5ed0ddb9360f8b762607996d12992 Mon Sep 17 00:00:00 2001
From: Salil Surendran <salilsurendran@cloudera.com>
Date: Mon, 27 Feb 2017 00:35:57 -0800
Subject: [PATCH 484/517] CDH-50687. Fix broken unit test

DiscoverySuite initializes TestHiveSingleton before the initial test run thereby creating a SparkContext which leads to errors when the test run starts and tries to create another duplicate SparkContext.
To ensure that all tests are using the same SparkContext make sure that FileQueryAnalysisSuite is also extended from TestHiveSingleton rather than SharedSQLContext.
---
 .../query/analysis/FileQueryAnalysisSuite.scala    |    5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/lineage/src/test/scala/org/apache/spark/sql/query/analysis/FileQueryAnalysisSuite.scala b/lineage/src/test/scala/org/apache/spark/sql/query/analysis/FileQueryAnalysisSuite.scala
index c3e3216..dd04ef6 100644
--- a/lineage/src/test/scala/org/apache/spark/sql/query/analysis/FileQueryAnalysisSuite.scala
+++ b/lineage/src/test/scala/org/apache/spark/sql/query/analysis/FileQueryAnalysisSuite.scala
@@ -19,6 +19,7 @@ package org.apache.spark.sql.query.analysis
 
 import org.apache.spark.SparkFunSuite
 import org.apache.spark.sql.execution.datasources.parquet.ParquetTest
+import org.apache.spark.sql.hive.test.TestHiveSingleton
 import org.apache.spark.sql.query.analysis.DataSourceType.DataSourceType
 import org.apache.spark.sql.query.analysis.TestUtils._
 import org.apache.spark.sql.test.SharedSQLContext
@@ -26,8 +27,8 @@ import org.apache.spark.sql.test.SharedSQLContext
 /**
  * Tests that reading and writing to the local and HDFS file systems produces the desired lineage.
  */
-class FileQueryAnalysisSuite extends SharedSQLContext with ParquetHDFSTest with
-ParquetTest {
+class FileQueryAnalysisSuite extends ParquetTest with ParquetHDFSTest with TestHiveSingleton
+ {
 
   test("Local file works") {
     testSimpleQuery(withParquetFile, DataSourceType.LOCAL)
-- 
1.7.9.5

